api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java

```diff
@@ -525,6 +527,66 @@ public void test_outdatedDataUseAgreement() {
     assertThat(accessTierService.getAccessTiersForUser(dbUser)).doesNotContain(registeredTier);
   }
 
+  @Test
+  public void test_syncAccessModuleStatus() {
+    config.featureFlags.enableV3DataUserCodeOfConduct = true;
+    config.accessRenewal.expiryDays = (long) 365;
+    config.access.enableAccessRenewal = true;
+
+    final long userId = createAccountAndDbUserWithAffiliation().getUserId();
+
+    // Set bypass times
+    final DbUser dbUser = userDao.findUserByUserId(userId);
+    dbUser.setBetaAccessBypassTime(TIMESTAMP);
+    dbUser.setEraCommonsBypassTime(TIMESTAMP);
+    dbUser.setTwoFactorAuthBypassTime(TIMESTAMP);
+
+    // Confirm the user does not have access yet
+    assertThat(accessTierService.getAccessTiersForUser(dbUser)).isEmpty();
+
+    // User completes access
+    dbUser.setDataUseAgreementCompletionTime(TIMESTAMP);
+    dbUser.setDataUseAgreementSignedVersion(DUA_VERSION);
+    dbUser.setComplianceTrainingCompletionTime(TIMESTAMP);
+    dbUser.setProfileLastConfirmedTime(TIMESTAMP);
+    dbUser.setPublicationsLastConfirmedTime(TIMESTAMP);
+    userDao.save(dbUser);
+    dbUser.getComplianceTrainingCompletionTime();
+
+    // We request a sync, the user is now compliant
+    assertThat(profileController.syncAccessModuleStatus().getStatusCode()).isEqualTo(HttpStatus.OK);
+    assertThat(accessTierService.getAccessTiersForUser(dbUser)).contains(registeredTier);
+
+    // // One year passes
+    long yearInMillis = TimeUnit.DAYS.toMillis(config.accessRenewal.expiryDays + 1);
+    fakeClock.increment(yearInMillis);
+   
+
+    // A sync is requested - it should remove access
+    assertThat(profileController.syncAccessModuleStatus().getStatusCode()).isEqualTo(HttpStatus.OK);
+    assertThat(accessTierService.getAccessTiersForUser(dbUser)).isEmpty();
+
+    // User updates everything but their profile
```

Comment:
RE keeping this test without the endpoint: I'd suggest keeping it here and switching these `setX` interactions here to instead use the controller APIs to recomplete modules.

The only  thing this won't handle well is the AAR expiration. You'd need to come up with another approach to trigger that recalculation from here.